pkg/codegen/python/gen.go

```diff
@@ -508,7 +577,13 @@ func (mod *modContext) importResourceFromToken(tok string) string {
 
 	refPkgName := parts[0]
 
-	modName := mod.tokenToResource(tok)
+	resName := ""
+	if refPkgName == "pulumi" && parts[1] == "providers" {
+		refPkgName, resName = parts[2], "Provider"
+		parts[0], parts[1], parts[2] = refPkgName, "", "Provider"
```

Comment:
Looks like bad merge with my change a few lines above. Happy to take this implementation, though, if you want to delete the lines above.

Reply:
Your change is good--I'll adjust this.